tools.yml

```
@@ -129,126 +106,1952 @@ tools:
+    mem: 92
+  toolshed.g2.bx.psu.edu/repos/galaxyp/cardinal_spectra_plots/cardinal_spectra_plots/.*:
+    mem: 32
+  toolshed.g2.bx.psu.edu/repos/galaxyp/cardinal_data_exporter/cardinal_data_exporter/.*:
+    mem: 200
+  toolshed.g2.bx.psu.edu/repos/ethevenot/heatmap/Heatmap/.*:
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/mitos2/mitos2/.*:
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/feelnc/feelnc/.*:
+    mem: 12
   toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/.*:
-    rules:
-    - id: large_job_rule
-      if: input_size >= 0.001
-      cores: 5
-  toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/.*:
-    cores: 2
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/.*:
+    cores: 10
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/jellyfish/jellyfish/.*:
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline/.*:
+    cores: 12
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/iuc/shasta/shasta/.*:
+    cores: 10
+    mem: 12
+  toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/.*:
     scheduling:
-      accept:
-      - pulsar
-    rules:
-    - id: medium_job_rule
-      if: 0.001 < input_size <= 5
-      cores: 16
-    - id: large_job_rule
-      if: input_size > 5
-      cores: 32 # MAX
-  toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/.*:
-    cores: 4
+      require:
+      - docker
+  toolshed.g2.bx.psu.edu/repos/iuc/lumpy_sv/lumpy_sv/.*:
+    mem: 8
+  toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/.*:
+    cores: 24
+    mem: 250
     scheduling:
-      accept:
-      - pulsar
-    rules:
-    - id: medium_job_rule
-      if: 0.5 <= input_size < 5
-      cores: 8
-    - id: large_job_rule
-      if: 5 <= input_size < 20
-      cores: 16
-    - id: xlarge_job_rule
-      if: input_size >= 20
-      cores: 32 # MAX
+      require:
+      - docker
+  toolshed.g2.bx.psu.edu/repos/bgruening/mitohifi/mitohifi/.*:
+    cores: 8
+    mem: 16
+    scheduling:
+      require:
+      - docker
+  toolshed.g2.bx.psu.edu/repos/rnateam/rnalien/RNAlien/.*:
+    cores: 10
+  toolshed.g2.bx.psu.edu/repos/iuc/slamdunk/slamdunk/.*:
+    cores: 10
+  toolshed.g2.bx.psu.edu/repos/iuc/pangolin/pangolin/.*:
+    cores: 8
   toolshed.g2.bx.psu.edu/repos/iuc/pilon/pilon/.*:
-    cores: 9
-    rules:
-    - id: large_job_rule
-      if: input_size >= 4
-      cores: 16
+    mem: 32
     env:
       _JAVA_OPTIONS: '-Xmx{int(mem)}G -Xms1G'
-  toolshed.g2.bx.psu.edu/repos/iuc/poretools_.*:  # very general match
-    cores: 3
-  toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/.*:
-    cores: 3
-  toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fastq_dump/.*:
-    cores: 3
-  toolshed.g2.bx.psu.edu/repos/iuc/trinity/trinity/.*:
+  toolshed.g2.bx.psu.edu/repos/galaxyp/encyclopedia_prosit_csv_to_library/encyclopedia_prosit_csv_to_library/.*:
+    mem: 40
+    env:
+      _JAVA_OPTIONS: '-Xmx{int(mem)}G -Xms1G'
```

Should we consider defining this once somewhere and inheriting it? Something like:
```suggestion
__java_tool:
    context:
        java_heap_min: 1
     env:
      _JAVA_OPTIONS: '-Xmx{int(mem)}G -Xms{int(java_heap_min)}G'
  toolshed.g2.bx.psu.edu/repos/galaxyp/encyclopedia_prosit_csv_to_library/encyclopedia_prosit_csv_to_library/.*:
    inherits: __java_tool
    mem: 40
```
Looks like it could help here: https://github.com/usegalaxy-eu/infrastructure-playbook/pull/401/files#diff-8db6ee20f85fcb05432e330fc9eda5abece14a8fa97cdfd3c5a4ed42f893d896R316

But not here: https://github.com/usegalaxy-eu/infrastructure-playbook/pull/401/files#diff-8db6ee20f85fcb05432e330fc9eda5abece14a8fa97cdfd3c5a4ed42f893d896R387

---

We could go further and add the `_JAVA_OPTIONS` to the default tool and only a few tools might override it